File: 0952-word-subsets/0952-word-subsets.cs
public class Solution {
    private bool helper(int[] f, int[] mf) {
        for (int u = 0; u < 26; ++u) {
            if (f[u] < mf[u])
                return false;
        }
        return true;
    }
    public IList<string> WordSubsets(string[] words1, string[] words2) {
        var ans = new List<string>();
        int[] mf = new int[26];
        foreach (var word in words2) {
            int[] f = new int[26];
            foreach (var ch in word)
                f[ch - 'a']++;
            for (int i = 0; i < 26; i++)
                mf[i] = Math.Max(mf[i], f[i]);
        }

        foreach (var word in words1) {
            int[] f = new int[26];
            foreach (var ch in word) 
                f[ch - 'a']++;
            
            if (helper(f, mf))
                ans.Add(word);
        }
        return ans;
    }
}

File: 0952-word-subsets/0952-word-subsets.cpp
class Solution {
private:
    bool helper(int f[], int mf[]) {
        for (uint i = 0; i < 26; i++) {
            if (f[i] < mf[i])
                return false;
        }
        return true;
    }
    
public:
    vector<string> wordSubsets(vector<string>& words1, vector<string>& words2) {
        vector<string> ans;
        int mf[26] = {};
        for (auto& word : words2) {
            int f[26] = {};
            for (auto& s : word)
                f[s - 'a']++;
            
            for (int i = 0; i < 26; ++i)
                mf[i] = fmax(mf[i], f[i]);
            
        }
        for (auto& word : words1) {
            int f[26] = {};
            for (auto& s : word)
                f[s - 'a']++;
            if (helper(f, mf))
                ans.push_back(word);
        }
        return ans;
    }
};